From d0e6f069b8776ef80373db8fa3bad03845e8400e Mon Sep 17 00:00:00 2001
From: Maulik Patel <maulik.patel@arm.com>
Date: Fri, 27 Jun 2025 12:58:35 +0100
Subject: [PATCH 5/5] BL2: Fix TFM CI failure due to FIH call

'boot_verify_key_id_for_image' is an FIH call causing the TFM BL2
tests to fail in CI. Update the return type to 'fih_ret' so the
fault injection control flow works correctly.

Signed-off-by: Maulik Patel <maulik.patel@arm.com>
Change-Id: If4690a863276ff9843874093009e0654620a2334
---
 boot/bootutil/include/bootutil/sign_key.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/boot/bootutil/include/bootutil/sign_key.h b/boot/bootutil/include/bootutil/sign_key.h
index 6e20ba0c..613f6776 100644
--- a/boot/bootutil/include/bootutil/sign_key.h
+++ b/boot/bootutil/include/bootutil/sign_key.h
@@ -28,6 +28,10 @@
 /* mcuboot_config.h is needed for MCUBOOT_HW_KEY to work */
 #include "mcuboot_config/mcuboot_config.h"
 
+#ifdef MCUBOOT_BUILTIN_KEY
+#include "bootutil/fault_injection_hardening.h"
+#endif /* MCUBOOT_BUILTIN_KEY */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -48,7 +52,7 @@ extern const struct bootutil_key bootutil_keys[];
  *
  * @return                   0 if the key ID is valid for the image; nonzero on failure.
  */
-int boot_verify_key_id_for_image(uint8_t image_index, uint32_t key_id);
+fih_ret boot_verify_key_id_for_image(uint8_t image_index, uint32_t key_id);
 #endif /* MCUBOOT_BUILTIN_KEY */
 #else
 struct bootutil_key {
-- 
2.34.1

